### ISSUE TITLE
This loop never actually loops (but it does)

### ISSUE THREAD

**Issue by Antosser** [rust-lang/rust-clippy#11007]
### Summary

```rust
fn main() {
    'outer: loop {
        'inner: {
            if true {
                break 'inner;
            }

            break 'outer;
        }
    }
}
```
This code produces the following Clippy error:
```
error: this loop never actually loops
  --> src\main.rs:2:5
   |
2  | /     'outer: loop {
3  | |         'inner: {
4  | |             if true {
5  | |                 break 'inner;
...  |
9  | |         }
10 | |     }
   | |_____^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#never_loop
   = note: `#[deny(clippy::never_loop)]` on by default
```
This shouldn't be an error, as the loop continues forever

### Lint Name

never_loop

### Reproducer

I tried this code:

```rust
<code>
```

I saw this happen:

```
<output>
```

I expected to see this happen:

### Version

```text
rustc 1.70.0 (90c541806 2023-05-31)
binary: rustc
commit-hash: 90c541806f23a127002de5b4038be731ba1458ca
commit-date: 2023-05-31
host: x86_64-pc-windows-msvc
release: 1.70.0
LLVM version: 16.0.2
```

**Comment by Antosser** [comment]
Duplicate https://github.com/rust-lang/rust-clippy/issues/11004
